api/runnerpool/runner_pool.go

```diff
@@ -42,7 +42,7 @@ type Runner interface {
 // processed by a RunnerPool
 type RunnerCall interface {
 	SlotDeadline() time.Time
-	Request() *http.Request
+	RequestBody() io.ReadCloser
```

Comment:
Probably in the future we need to use some deprecation process before changing a public interface

Reply:
Agreed on that. I'd like to go through the interfaces here once more to ensure they are as small as possible; there's no point requiring someone to construct an httpRequest if all we need is the body stream, for instance.